neqo-transport/src/connection.rs

```diff
@@ -2614,4 +2617,55 @@ mod tests {
         assert_eq!(evts.len(), 1);
         assert!(matches!(evts[0], ConnectionEvent::SendStreamWritable{..}));
     }
+
+    // Test that we split crypto data if they cannot fit into one packet.
+    // To test this we will use a long server certificate.
+    #[test]
+    fn test_crypto_frame_split() {
+        let mut client = default_client();
+        let mut server = Connection::new_server(
+            test_fixture::LONG_CERT_KEYS,
+            test_fixture::DEFAULT_ALPN,
+            &test_fixture::anti_replay(),
+        )
+        .expect("create a server");
+
+        let client1 = client.process(None, now());
+        assert!(client1.as_dgram_ref().is_some());
+
+        // The entire server flight doesn't fit in a single packet because the
+        // certificate is large, therefore the server will produce 2 paclets.
+        let server1 = server.process(client1.dgram(), now());
+        assert!(server1.as_dgram_ref().is_some());
+        let server2 = server.process(None, now());
+        assert!(server2.as_dgram_ref().is_some());
+
+        let client2 = client.process(server1.dgram(), now());
+        assert!(client2.as_dgram_ref().is_some()); // this is an ack.
+        let _ = maybe_authenticate(&mut client); // maybe client has the cert but will not have complete server fight to finish handshate.
+        assert_eq!(*client.state(), State::Handshaking);
+
+        // let server process the ack for the first packet.
+        let server3 = server.process(client2.dgram(), now());
+        assert!(server3.as_dgram_ref().is_none());
+
+        // Consume the second packet from the server.
+        let client3 = client.process(server2.dgram(), now());
+
+        // check authentication.
+        let _ = maybe_authenticate(&mut client);
```

Comment:
You should maybe save the two responses and then:

```suggestion
        let auth2 = maybe_authenticate(&mut client);
        assert!(auth1 ^ auth2);
```

Just so that we have a test that ensures we don't trigger the authentication event twice.